src/wp-includes/blocks.php

```diff
@@ -1009,6 +1009,24 @@ function excerpt_remove_blocks( $content ) {
 	return $output;
 }
 
+/**
+ * Parses footnotes markup out of a content string,
+ * and renders those appropriate for the excerpt.
+ *
+ * @since 6.3.0
+ *
+ * @param string $content The content to parse.
+ * @return string The parsed and filtered content.
+ */
+function excerpt_remove_footnotes( $content ) {
+	if ( false === strpos( $content, 'data-fn=' ) ) {
+		return $content;
+	}
+
+	static $footnote_pattern = '_<sup data-fn="[^"]+" class="[^"]+">\s*<a href="[^"]+" id="[^"]+">\d+</a>\s*</sup>_';
```

Comment:
Does this need to be stored in a variable rather than passed directly to `preg_replace()`?

Reply:
Oh good call. I blindly copied from https://github.com/WordPress/gutenberg/pull/52518